docs/development/machine-bootstrap.md

```diff
@@ -0,0 +1,22 @@
+# Machine-Controller integration into Gardener
+The Gardener deploys the [machine-controller-manager](https://github.com/gardener/machine-controller-manager) as part of a Shoot controlplane onto a Seed to manage the nodes/machines of the Shoot cluster. The machine-controller-manager will create, bootstrap, update and delete the nodes of the Shoot cluster and the underlying virtual machines.
+
+## Flow of creating and bootstrapping Machines/Nodes
+The following steps will be ensured by the garden-controller-manager during a creation/reconciliation flow:
+
+- The machine-controller-manager gets deployed into the Shoot namespace of the Seed.
+- The proper ``<cloudprovider>MachineClass`` objects and their corresponding secrets gets deployed in the Shoot namespace of the Seed. Those secrets contain information how to bootstrap the nodes and will be used by the machine-controller-manager to configure cloud-config/user-data of the VM.
+- The ``MachineDeployment``s for the worker pools specified in the Shoot manifest will be applied. Those deployments will result in the required ``Machine`` objects in the Shoot namespace of the Seed.
+- A secret, which contain a valid bootstrap token and
+the cloud-config configuration for bootstrapping nodes, will be created in the kube-system namespace of the Shoot.
+
+The machine-controller-manager will pick up the ``Machine`` objects and create the VMs on the respective cloud provider. Bootstrapping VMs is done via OS-specific configuration (e.g., [CoreOS cloud-config]((https://coreos.com/os/docs/latest/cloud-config.html))).
```

Comment:
We already talk about the fact that the bootstrapping is happening via cloud config, but we only explain it here (later). Let's move up this paragraph.